24.md

```diff
@@ -0,0 +1,219 @@
+7. **No Shared Secrets**: There is no chatroom secret key that can leak or be mistakenly shared
+8. **Fully Recoverable**: Messages can be fully recoverable in any client with the receiver or the sender's private key
+9. **Optional Forward Secrecy**: Users and clients can opt-in for "disappearing messages" that are not recoverable with their private key
+10. **Uses Public Relays**: The protocol's messages can flow through public relays without loss of privacy. Private relays can increase privacy further, but they are not needed.
+11. **Cold Storage**: Users will be able to also opt-in to sharing messages with a new key exclusive for DM backup and recovery.
+11. **Composable**: The protocol is extensible to make any other event kind fully private (private likes, private reports, private long-form content, etc)
+
+## 1. The Unsigned Direct Message
+
+A new `kind:14` defines a Chat Message between a group of users (i.e. pubkey + p-tags). 
+
+It is designed to be a Rumor in [NIP-59](https://github.com/nostr-protocol/nips/pull/716/files) and thus MUST never be signed. If it is signed, the information in it is **fully public**. This event kind requires the rest of the protocol to make it private. 
+
+```js
+{
+  "id": "<The usual hash>",
+  "pubkey": "<author key in hex>",
+  "content": "This is a DM",
+  "kind": 14,
+  "created_at": 1686840217,
+  "tags": [ <one or more receivers, mentions, citations, additional fields like content-sensitivity, etc> ],
+}
+```
+
+The `.content` contains the message. 
+
+The set of `pubkey` + `p` tags define a Group. Clients SHOULD render messages with the same group in a continuous thread. If a new `p` tag is added, a new group is created. 
+
+An optional `subject` tag defines the current name/topic of the conversation. Any member can change the topic by simply submitting a new `subject` to an existing `pubkey` + `p`-tags group. 
+
+The use of other tags is highly encouraged (e.g. such as `content-warning` ([NIP-36](36.md)) if the content is NSFW). 
+
+## 2. The Seal
+
+A Seal is a `kind:13` event that authenticates the author of the encrypted message. It wraps the **unsigned** `kind:14` event (the Rumor) and is signed with the sender's regular key. 
+
+```js
+{
+  "id": "<The usual hash>",
+  "pubkey": "<Author's Main PubKey>",
+  "content": "<XCHACHA-Encrypted Base64 of the Kind 14>",
+  "kind": 13,
+  "created_at": 1686840217,			// No need to match the current time
+  "tags": [],   					// Always empty
+  "sig": "<Author's Main PubKey Signature>"
+}
+```
+
+The Sealed Rumor is **always** encrypted to a receiver's pubkey but **no** `p` tag pointing to the receiver is used. The receiver's information is not public. The only public information in this event is who is signing it. 
+
+Clients MUST verify if pubkey of the `kind:13` is the pubkey on the `kind:14`, otherwise any author can impersonate others by simply changing the pubkey on `kind:14`.
+
+Clients SHOULD randomize `created_at` to reduce time-collisions.
```

Comment:
Randomize is too vague. Over what range? Should i use times from 1972? Probably would be rejected by relays. Maybe "Clients SHOULD use a low-precision `created_at` that differs from <what?> to make time correlation of metadata less effective"

Reply:
It's hard to pin a range because it's dependent on how many GiftWraps are out there. Today Amethyst is randomizing within the past week. I believe Coracle does the same. But if we had lots of use, we wouldn't need to go that far.

Reply:
This should probably just be removed, it's already in NIP 59 with more specificity.